modules/misunderstood/src/backend/db.ts

```diff
@@ -115,10 +115,15 @@ export default class Db {
 
     const context = [...messagesBefore, ...messagesAfter]
       .sort((e1, e2) => e1.id - e2.id)
-      .map(({ id, event }) => ({
-        ...pick(typeof event === 'string' ? JSON.parse(event) : event, 'preview', 'direction'),
-        isCurrent: id === messageId
-      }))
+      .map(({ id, event }) => {
+        const eventObj = typeof event === 'string' ? JSON.parse(event) : event
+        return {
+          direction: eventObj.direction,
+          preview: (eventObj.preview || '').replace(/<[^>]*>?/gm, ''),
+          payloadMessage: get(eventObj, 'payload.message'),
+          isCurrent: id === messageId
+        }
```

Comment:
You can extract the event from the database as JSON without using JSON.parse. Have a look at the event collector repository: https://github.com/botpress/botpress/blob/master/src/bp/core/repositories/events.ts#L52

Reply:
I agree, but this part didn't change here.